mlir/lib/Gradient/Transforms/gradient_to_llvm.cpp

```diff
@@ -57,6 +58,17 @@ struct GradientConversionPass : impl::GradientConversionPassBase<GradientConvers
         target.addLegalDialect<arith::ArithDialect, linalg::LinalgDialect, func::FuncDialect,
                                index::IndexDialect, memref::MemRefDialect>();
 
+        // This is a bit unfortunate.
```

Comment:
Maybe not that unfortunate, please verify

Reply:
This is also not possible. Because the function to be derived is a catalyst.callback and the function that lowers the GradTriple will generate `func.constant` and `func.constant` can only use `func.func` symbols.

Reply:
Maybe a downside after all to create all these operations which are effectively functions? 😅 Either way I think this is okay for now, the issue is well documented 👍